src/Disciplina/Educator/SizedMerkleTree.hs

```diff
@@ -0,0 +1,237 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE DeriveFoldable #-}
+{-# LANGUAGE DeriveAnyClass #-}
+
+-- | Sized Merkle tree implementation.
+module Disciplina.Educator.SizedMerkleTree
+       ( SizedMerkleRoot(..)
+       , SizedMerkleTree (..)
+       , getSizedMerkleRoot
+       , mkSizedMerkleTree
+
+       , SizedMerkleNode (..)
+       , mkBranch
+       , mkLeaf
+       ) where
+
+import Universum hiding (foldMap, toList)
+
+import Disciplina.Crypto (Hash, HasHash, hash, unsafeHash)
+import Disciplina.Crypto.Hash.Class (HashFunc, AbstractHash (..))
+
+import Codec.Serialise (Serialise(..), serialise)
+import Data.Bits (Bits (..))
+import Data.ByteArray (ByteArrayAccess, convert)
+import Data.ByteString.Builder (Builder, byteString, lazyByteString)
+import Data.Foldable (Foldable (..))
+import Prelude (show)
```

Comment:
What's wrong with the universum's `show`? If you need to define your own `Show` instance, just `import qualified Prelude`.